vector_setters.go

```diff
@@ -175,5 +429,75 @@ func (vec *vector) setMap(rowIdx C.idx_t, val any) {
 		i++
 	}
 
-	vec.setList(rowIdx, list)
+	setList(vec, rowIdx, list)
+	return nil
+}
+
+func setUUID[S any](vec *vector, rowIdx C.idx_t, val S) error {
+	var uuid UUID
+	switch v := any(val).(type) {
+	case UUID:
+		uuid = v
+	default:
+		return castError(reflect.TypeOf(val).String(), reflect.TypeOf(uuid).String())
+	}
+	hi := uuidToHugeInt(uuid)
+	setPrimitive(vec, rowIdx, hi)
+	return nil
+}
+
+// TODO: Support ARRAY UNION BIT ANY VARINT,
+func setVectorVal[S any](vec *vector, rowIdx C.idx_t, val S) error {
+	switch vec.Type {
+	case TYPE_BOOLEAN:
+		return setBool[S](vec, rowIdx, val)
+	case TYPE_TINYINT:
+		return setNumeric[S, int8](vec, rowIdx, val)
+	case TYPE_SMALLINT:
+		return setNumeric[S, int16](vec, rowIdx, val)
+	case TYPE_INTEGER:
+		return setNumeric[S, int32](vec, rowIdx, val)
+	case TYPE_BIGINT:
+		return setNumeric[S, int64](vec, rowIdx, val)
+	case TYPE_UTINYINT:
+		return setNumeric[S, uint8](vec, rowIdx, val)
+	case TYPE_USMALLINT:
+		return setNumeric[S, uint16](vec, rowIdx, val)
+	case TYPE_UINTEGER:
+		return setNumeric[S, uint32](vec, rowIdx, val)
+	case TYPE_UBIGINT:
+		return setNumeric[S, uint64](vec, rowIdx, val)
+	case TYPE_FLOAT:
+		return setNumeric[S, float32](vec, rowIdx, val)
+	case TYPE_DOUBLE:
+		return setNumeric[S, float64](vec, rowIdx, val)
+	case TYPE_TIMESTAMP, TYPE_TIMESTAMP_S, TYPE_TIMESTAMP_MS,
+		TYPE_TIMESTAMP_NS, TYPE_TIMESTAMP_TZ:
+		return setTS[S](vec, vec.Type, rowIdx, val)
+	case TYPE_DATE:
+		return setDate[S](vec, rowIdx, val)
+	case TYPE_TIME:
+		return setTime[S](vec, rowIdx, val)
+	case TYPE_INTERVAL:
+		return setInterval[S](vec, rowIdx, val)
+	case TYPE_HUGEINT:
+		return setHugeint[S](vec, rowIdx, val)
+		// UHUGEINT is not supported
+	case TYPE_VARCHAR:
+		return setString[S](vec, rowIdx, val)
+	case TYPE_BLOB:
+		return setString[S](vec, rowIdx, val)
+	case TYPE_DECIMAL:
+		return setDecimal[S](vec, rowIdx, val)
+	case TYPE_ENUM:
+		return setEnum[S](vec, rowIdx, val)
+	case TYPE_LIST:
+		return setList[S](vec, rowIdx, val)
+	case TYPE_STRUCT:
+		return setStruct[S](vec, rowIdx, val)
+	case TYPE_UUID:
+		return setUUID[S](vec, rowIdx, val)
+	default:
+		return unsupportedTypeError("Could not convert type " + reflect.TypeOf(val).Name() + " to a duckdb type")
+	}
```

Comment:
This function is currently unused, so we should remove it to avoid adding untested/dead code.

Reply:
Sure, I can also add a generic set methods to the chunk API? Since that is what it is intended for as well

Reply:
Either that (plus the tests) or you can add it with the table UDFs.